gobblin-metrics/src/main/java/gobblin/metrics/MetricReportReporter.java

```diff
@@ -0,0 +1,304 @@
+   * Uses the serialize* methods in {@link gobblin.metrics.MetricReportReporter}.
+   * @param gauges
+   * @param counters
+   * @param histograms
+   * @param meters
+   * @param timers
+   */
+  @Override
+  public void report(SortedMap<String, Gauge> gauges, SortedMap<String, Counter> counters,
+      SortedMap<String, Histogram> histograms, SortedMap<String, Meter> meters, SortedMap<String, Timer> timers,
+      Map<String, String> tags) {
+
+    List<Metric> metrics = new ArrayList<Metric>();
+
+    for( Map.Entry<String, Gauge> gauge : gauges.entrySet()) {
+      metrics.addAll(serializeGauge(gauge.getKey(), gauge.getValue()));
+    }
+
+    for ( Map.Entry<String, Counter> counter : counters.entrySet()) {
+      metrics.addAll(serializeCounter(counter.getKey(), counter.getValue()));
+    }
+
+    for( Map.Entry<String, Histogram> histogram : histograms.entrySet()) {
+      metrics.addAll(serializeSnapshot(histogram.getKey(), histogram.getValue().getSnapshot()));
+      metrics.addAll(serializeSingleValue(histogram.getKey(), histogram.getValue().getCount(), "count"));
+    }
+
+    for ( Map.Entry<String, Meter> meter : meters.entrySet()) {
+      metrics.addAll(serializeMetered(meter.getKey(), meter.getValue()));
+    }
+
+    for ( Map.Entry<String, Timer> timer : timers.entrySet()) {
+      metrics.addAll(serializeSnapshot(timer.getKey(), timer.getValue().getSnapshot()));
+      metrics.addAll(serializeMetered(timer.getKey(), timer.getValue()));
+      metrics.addAll(serializeSingleValue(timer.getKey(), timer.getValue().getCount(), "count"));
+    }
+
+    Map<String, String> allTags = Maps.newHashMap();
+    allTags.putAll(tags);
+    allTags.putAll(this.tags);
+
+    MetricReport report = new MetricReport(allTags, new DateTime().getMillis(), metrics);
+
+    pushReport(report);
+
+  }
+
+  /**
+   * Emit the {@link gobblin.metrics.MetricReport} to the metrics sink.
+   * @param report metric report to emit.
+   */
+  protected abstract void pushReport(MetricReport report);
+
+  /**
+   * Extracts metrics from {@link com.codahale.metrics.Gauge}.
+   * @param name
+   * @param gauge
+   * @return a list of {@link gobblin.metrics.Metric}.
+   */
+  protected List<Metric> serializeGauge(String name, Gauge gauge) {
+    List<Metric> metrics = new ArrayList<Metric>();
+    try {
+      metrics.add(new Metric(name, Double.parseDouble(gauge.getValue().toString())));
+    } catch(NumberFormatException exception) {
+      LOGGER.info("Failed to serialize gauge metric. Not compatible with double value.");
+    }
+    return metrics;
+  }
+
+  /**
+   * Extracts metrics from {@link com.codahale.metrics.Counter}.
+   * @param name
+   * @param counter
+   * @return a list of {@link gobblin.metrics.Metric}.
+   */
+  protected List<Metric> serializeCounter(String name, Counter counter) {
+    List<Metric> metrics = new ArrayList<Metric>();
+    metrics.add(serializeValue(name, counter.getCount()));
+    return metrics;
```

Comment:
You can do `return Lists.newArrayList(serializeValue(name, counter.getCount()));`.